python/lsst/daf/butler/_butler.py

```diff
@@ -281,6 +281,47 @@ def __init__(self, config: Union[Config, str, None] = None, *,
     code.
     """
 
+    @classmethod
+    def get_repo_uri(cls, label: str) -> ButlerURI:
+        """Look up the label in a butler repository index.
+
+        Parameters
+        ----------
+        label : `str`
+            Label of the Butler repository to look up.
+
+        Returns
+        -------
+        uri : `ButlerURI`
+            URI to the Butler repository associated with the given label.
+
+        Raises
+        ------
+        KeyError
+            Raised if the label is not found in the index, or if an index
+            can not be found at all.
+
+        Notes
+        -----
+        The index of butler repositories is found by looking for a
+        configuration file at the URI pointed at by the environment
+        variable ``$BUTLER_REPOSITORY_INDEX``. The configuration file
+        is a simple dictionary lookup of the form:
+
+        .. code-block:: yaml
+
+           label1: uri1
+           label2: uri2
+
+        and can be in YAML or JSON format.
+        """
+        index_uri = os.environ.get("BUTLER_REPOSITORY_INDEX")
+        if index_uri is None:
+            raise KeyError("No repository index defined in enviroment variable BUTLER_REPOSITORY_INDEX.")
+
+        config = Config(index_uri)
```

Comment:
Could it be worth saving this rather than reloading it on every query?  Most of the time there would be exactly one call to this function, but I could see some cases where there might be more than one.  Also, I think it might be useful to provide a method to list all the labels to enable the user to be asked to select one.

Reply:
`Butler.list_repo_labels()` ?

Reply:
I've reworked things such that there is now a `ButlerRepoIndex` class that does the heavy lifting and the two `Butler` methods forward directly to that class.  The main issue is whether we bother with the `Butler` class methods or just tell people they have to use `ButlerRepoIndex` directly. The point in favor of `Butler` is that they already have that class imported. The point against is that it's new methods cluttering up the class that are only tangentially related to `Butler`. @mfisherlevine do you have an opinion on:

```python
from lsst.daf.butler import Butler
uri = Butler.get_repo_uri("latiss")
butler = Butler(uri, ...)
```
vs
```python
from lsst.daf.butler import Butler, ButlerRepoIndex
uri = ButlerRepoIndex.get_repo_uri("latiss")
butler = Butler(uri, ...)
```
?